libs/domain/picking/offline/services/picking-sync-action-handler.service.ts

```diff
@@ -83,15 +83,14 @@ export class PickingSyncActionHandlerService
     return this.onlineAdapter.get({ ids: sync.payload.ids }).pipe(
       combineLatestWith(this.indexedDbService.getStore(PickingListEntity)),
       switchMap(async ([pickingLists, store]) => {
-        const addedOrUpdatedKeys = await store.bulkPut(pickingLists, {
-          allKeys: true,
+        const pickingListsIdsToRemove = sync.payload.ids.filter((id) => {
+          return !pickingLists.find((pl) => pl.id === id);
         });
 
-        if (addedOrUpdatedKeys.length !== pickingLists.length) {
```

Comment:
We don't need this check anymore?

Reply:
I think there is no reason to check the number `addedOrUpdatedKeys` because we trust Dexie methods that items will be added/updated/removed